src/main/java/com/google/codeu/servlets/CinemaInitServlet.java

```diff
@@ -0,0 +1,67 @@
+package com.google.codeu.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.Marker;
+import java.io.IOException;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonElement;
+import com.google.codeu.utils.CsvLoader;
+
+/** Handles initialization of {@link Marker} instances for each cinema. */
+@WebServlet("/cinema-init")
+public class CinemaInitServlet extends HttpServlet {
+
+  private static String cinemaLocationFilePath = "/WEB-INF/cinemaLocations.csv";
+  private Datastore datastore;
+  private JsonArray cinemaInfos;
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  /**
+   * Checks user login credentials. If user is admin, loads cinema location and description from
```

Comment:
hi jiang yue, this kind of differs from the UI sketches I've uploaded in the shared drive (but i get that its probably because I didn't upload in time). But right now the UI that I've implemented is that all users can go to the maps page (when they click on the ticket button from the landing page). When users want to post a request (when he or she clicks the map marker), he or she would enter: User ID, enter review, upload img (optional). Maybe my sketches aren't so clear so I'd either draw out a more detailed sketch or type out a functional document asap.

Reply:
oh oops i realized this is about adding a new cinema location and not adding a new review

Reply:
@jiangyue12392 @Reine0017 @Meg1456 Let's try to maintain a good description for PRs, that will hopefully fix this misunderstanding of PRs' purpose. In descriptions, lets include:

1. What you are changing/adding/removing
2. Why you are changing it
3. Any relevant supporting documents/github issues/conversations/mocks
4. Optional: What it was before, if that is unclear

Reply:
@Reine0017 The user shouldn't need to enter its own userID. The user should only be able to comment if already logged in and the userID should be automatically retrieved from login credentials.

Reply:
Here's the reasons why I'm combining "login" with entering a review:

1) Right now, user only needs to enter the name (email) to log in. Design wise it doesn't make much sense to have a page or form dedicated to logging in when its just one field.
2) In terms of usage, it makes sense that most of the time a user will just enter one review each time. Even if the user wants to enter more than one review, it wouldn't be much trouble to just enter his or her name again in the submission field.
3) All users should be able to view reviews and content even if not logged in, like most review sites or forums.

This is more of the frontend stuff, and your current version of the features is quite different from the sketches I've put up online. Perhaps the version I uploaded wasn't clear so I'd probably reupload another sketch. Of course the feature design is open for discussion and changes but it'll be great if these things can be raised in a central place so we are all on the same page.

Reply:
okkay, thanks for explaining, will redo UI sketch :)